src/compiler/parser.ts

```diff
@@ -1124,6 +1124,14 @@ namespace ts {
             return token === t && tryParse(nextTokenCanFollowModifier);
         }
 
+        function nextTokenIsOnSameLineAndCanFollowModifier() {

```

Comment:
It was `nextTokenCanFollowModifierWithoutGrammarRules` before which seemed more clear to me. Basically, I just need `nextTokenCanFollowModifier` without the leading checks. I can think of several solutions here:

1. Inline `nextTokenIsOnSameLineAndCanFollowModifier` to `parseModifiers` (It leads to code duplication).

2. Rename `nextTokenIsOnSameLineAndCanFollowModifier` and use it in `nextTokenCanFollowModifier` as now.

3. Somehow pass a flag to `tryParse`. It seems like a big change because `tryParse` is a generic function that is used everywhere.

The option 2 seems reasonable overall.